File: python/insertion_sort.py
#!/usr/bin/env python3
from tools import common_funcs

# define global vars
path_to_shared_vars = 'tools/common_vars.json'
unsorted_array = common_funcs.read_array_from_json_file(file_path=path_to_shared_vars, array_name="my_array")
unsorted_array_len = len(unsorted_array)

# start at 1 to skip first value
for i in range(1, unsorted_array_len):
    temp = unsorted_array.pop(i)
    last_idx_of_sorted = i - 1

    # Go through sorted array
    while temp < unsorted_array[last_idx_of_sorted]:
        
        # update
        last_idx_of_sorted -= 1

        # Once it gets to last element, break
        if last_idx_of_sorted == -1:
            break
    
    # Place temp value within sorted array
    unsorted_array.insert(last_idx_of_sorted+1, temp)

print(unsorted_array)

    

File: cpp/src/insertion_sort.cpp
#include <iostream>
#include "../include/common_vars.h"

int main() {

    // Start at 1 because want to skip the first element
    for (int i = 1; i < my_vector_size; i++){

        // Remove value from vector and store in temp
        int temp = my_vector[i];
        my_vector.erase(my_vector.begin()+i);

        // Start checking from the end of the sorted array
        int left_idx_of_sorted = i - 1;

        // Go through sorted array
        while (temp < my_vector[left_idx_of_sorted]){

            // Update iterator of sorted array
            left_idx_of_sorted -= 1;
        }

        // Place temp value in sorted array
        auto it = my_vector.begin() + (left_idx_of_sorted+1);
        my_vector.insert(it, temp);
        
    }

    print_int_vector(my_vector);

    return 0;
    
    }